### PR TITLE
[Enhancement] Add full clone and version GC for PseudoCluster

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

This PR add full clone and version GC functionalities to PseudoCluster, it also fixes a bug in PseudoCluster query handling and a bug in the new publish mechanism's TransactionChecker init timing.
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 25 (88.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 25 (88.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 22 / 25 (88.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by decster** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 25 / 28 (89.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 19 / 22 (86.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|6|9|66.67%|[908, 910, 914]|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/TabletQuorumFailedException.java|2|2|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoBackend.java [hunk 1]
```diff
@@ -784,23 +838,23 @@ public Future<PCancelPlanFragmentResult> cancelPlanFragment(
...
-                result.queryStatistics = pQueryStatistics;
-                result.eos = true;
-                return result;
-            });
+                result.status = statusPB(TStatusCode.INTERNAL_ERROR, "no queryId found for finstId " + fid);
+                return CompletableFuture.completedFuture(result);
+            }
+            final QueryProgress progress = queryProgresses.get(queryId);
+            if (progress == null) {
+                LOG.warn("no progress found for finstId {} queryId", fid, queryId);
```
Comment (chaoyli) [inline fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoBackend.java:851]: What's the meaning of finstId?

**Review by chaoyli — APPROVED** [review]

File: fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoBackend.java (hunk 1 above)
Comment (decster) [reply fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoBackend.java:851]: `fragment_instance_id`

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by decster** [comment]
Fixes #9515

**Comment by decster** [comment]
Fixes #10019

**Pull request merged**